expected/wasm32-wasi/defined-symbols.txt

```diff
@@ -296,17 +296,21 @@ __wasi_sock_send
 __wasi_sock_shutdown
 __wasilibc_access
 __wasilibc_cwd
+__wasilibc_deinitialize_environ
```

Comment:
Why bother with the laziness at all.  i.e. why not just call this `__wasilibc_reinitialize_environ`?

Reply:
Another way of putting it: seems that any program that calls `__wasilibc_deinitialize_environ` would already be explicitly opting into using environment variables wouldn't it?  So why delay re-initialization in that case, especially when it adds more complexity to our already pretty complex weak symbol dance.

Reply:
In the wizer use case, we want to de-initialize the environment before serializing the program state, and re-initialize it only after the program is deserialized and resumed. And, in the resumption case, to speed up startup time in the case that the program doesn't use any environment variables right away, it's nice for the re-initialization to be lazy.

Reply:
This seems rather specific to the wizer use case.   I'm not particularly worried about this one minor change, but in general it seems a little backwards to have contort libc for specific use case like this. 

IIUC the suspending and resuming thing that wizer is doing is supposed to be mostly transparent, so I would hope that there were not to many more changes like this needed, especially in libc.

The approach in general seem rather error prone since a program could read an environment variable during its startup phase than would end be our-of-date when the snapshot was loaded..  I guess those are just the fundamental risks of using such snapshotting?

Reply:
Yes, it is specific to the wizer use case. And yes, I expect this is just a fundamental risk of using such snapshotting.

I do expect this is fairly limited in scope. Libc doesn't cache much state, and most of what it does have, such as the malloc heap, the atexit list, `errno`, the `strtok` buffer, `stdin`/`stdout`/`stderr`, etc., isn't really desirable to reset. The one thing I can think of that we may want to be able to reset is the preopen table.